iterator.go

```diff
@@ -259,3 +259,74 @@ func (iter *Iterator) Error() error {
 func (iter *Iterator) IsFast() bool {
 	return false
 }
+
+// NodeIterator is an iterator for nodeDB to traverse a tree.
```

Comment:
can we add a bit more on the godoc, is the caller responsible for anything, etc..